background/adapter/deluge.js

```diff
@@ -0,0 +1,57 @@
+if (typeof torrentToWeb === 'undefined') {
+    var torrentToWeb = {};
+};
+
+if (typeof torrentToWeb.adapter === 'undefined') {
+    torrentToWeb.adapter = {};
+};
+
+torrentToWeb.adapter.deluge = function(baseUrl, username, password, autostart)
+{
+    var baseUrlObject = new URL(baseUrl);
+    baseUrlObject.username = username;
+    baseUrlObject.password = password;
+    baseUrlObject.pathname = '/json';
+    baseUrlObject.search = '';
+    baseUrlObject.hash = '';
+
+    baseUrl = baseUrlObject.toString();
+
+    return {
+        send: function(filename, data, callback)
+        {
+            var fileReader = new FileReader();
+            fileReader.addEventListener('load', function(){
+                var requestData = {
+                    'id': '2',
```

Comment:
Why exactly id "2"? Does this one have a special meaning?

Reply:
Okay, I figured out what this is about. Basically the response will include the same ID again, which is good for async environments, but since we don't care about the responses, we can completely ignore that (AKA, this is fine).